tests/e2e/support/objects/app-admin-settings/spaces/actions.ts

```diff
@@ -50,38 +51,58 @@ export const changeSpaceQuota = async (args: {
   ])
 }
 
-export const disableSpace = async (args: { page: Page; id: string }): Promise<void> => {
-  const { page, id } = args
-  await page.locator(util.format(spaceIdSelector, id)).click()
+export const disableSpace = async (args: {
+  page: Page
+  id: string
+  context: string
+}): Promise<void> => {
+  const { page, id, context } = args
+  const isBatchActions = context === 'batch-actions'
+
+  if (!isBatchActions) {
+    await page.locator(util.format(spaceIdSelector, id)).click()
+  }
   await page.waitForSelector(disableActionBtn)
-  await page.locator(disableActionBtn).click()
+  await page.locator(`.${context}`).locator(disableActionBtn).click()
   await page.waitForSelector(modalConfirmBtn)
 
   await Promise.all([
     page.waitForResponse(
       (resp) =>
-        resp.url().endsWith(encodeURIComponent(id)) &&
+        (isBatchActions || resp.url().endsWith(encodeURIComponent(id))) &&
         resp.status() === 204 &&
         resp.request().method() === 'DELETE'
     ),
     page.locator(modalConfirmBtn).click()
   ])
 }
 
-export const deleteSpace = async (args: { page: Page; id: string }): Promise<void> => {
-  const { page, id } = args
-  await page.locator(util.format(spaceIdSelector, id)).click()
+export const deleteSpace = async (args: {
+  page: Page
+  id: string
+  context: string
+}): Promise<void> => {
+  const { page, id, context } = args
+  const isBatchActions = context === 'batch-actions'
+  if (!isBatchActions) {
+    await page.locator(util.format(spaceIdSelector, id)).click()
+  }
   await page.waitForSelector(deleteActionBtn)
-  await page.locator(deleteActionBtn).click()
+  await page.locator(`.${context}`).locator(deleteActionBtn).click()
   await page.waitForSelector(modalConfirmBtn)
 
   await Promise.all([
     page.waitForResponse(
       (resp) =>
-        resp.url().endsWith(encodeURIComponent(id)) &&
+        (isBatchActions || resp.url().endsWith(encodeURIComponent(id))) &&
```

Comment:
Same question here as before from @JammingBen 
Why is `resp.url().endsWith(encodeURIComponent(id)))` not needed in case of batch actions?

Reply:
Now I know: In case of batch actions, we would need to wait for all IDs that have been deleted. While possible, it's not really needed here because `resp.status() === 204 && resp.request().method() === 'DELETE'` is sufficient, no?